models/Shift.js

```diff
@@ -17,12 +18,27 @@ const ShiftSchema= new mongoose.Schema({
   },
   wage: Number,
   tips: Boolean,
-  date: Date,
-  start_time: String,
-  end_time: String,
-  activeStatus: { type: Boolean, default: true }, 
+  start_at: Date,
+  end_at: Date,
+  activeStatus: { type: Boolean, default: true },
   more: String,
   availability: [ { type: String } ]
 });
 
+ShiftSchema.virtual('duration').
+ get(function() {

```

Comment:
This function should probably be more generic. I would just return the result of `date.subtract(new Date(this.end_at), new Date(this.start_at))`, and let whatever function is using this property handle the rest.

I don't understand why you need line 31. It looks like this is a question of formatting, which should probably just be handled in the template or controller.

Reply:
If I don't convert my length format into hours in my model, I'll have to loop through all my shifts to do it in my controller. Doesn't it increase procedure steps and time for data processing?
I used `Math.round( shiftLength * 100 ) / 100` to round my number to 2nd decimal place. So it won't be displayed as a long line of numbers in the DOM.